projects/RabbitMQ.Client/client/impl/AutorecoveringConnection.Recovery.cs

```diff
@@ -213,7 +213,7 @@ private bool TryRecoverConnectionDelegate()
 
         private void RecoverExchanges(RecoveryChannelFactory recoveryChannelFactory)
         {
-            foreach (var recordedExchange in _recordedExchanges.Values.Where(x => _config.TopologyRecoveryFilter?.ExchangeFilter(x) ?? true))
+            foreach (var recordedExchange in _recordedExchanges.Values.Where(x => _config.TopologyRecoveryFilter?.ExchangeFilter(x) ?? true).ToArray())
```

Comment:
Intended?

Reply:
Yeah. The foreach throws an exception if the collection being iterated over changes while it's running so I need a snapshot of it since it might take long.